cm15/codegen_client.go

```diff
@@ -8807,6 +8807,211 @@ func (loc *RepositoryAssetLocator) Show(options rsapi.APIParams) (*RepositoryAss
 	return res, err
 }
 
+/******  ResourceGroup ******/
+
+type ResourceGroup struct {
+	Actions     []map[string]string `json:"actions,omitempty"`
+	CreatedAt   *RubyTime           `json:"created_at,omitempty"`
+	Description string              `json:"description,omitempty"`
+	Links       []map[string]string `json:"links,omitempty"`
+	Name        string              `json:"name,omitempty"`
+	ResourceUid string              `json:"resource_uid,omitempty"`
+	State       string              `json:"state,omitempty"`
+	UpdatedAt   *RubyTime           `json:"updated_at,omitempty"`
+}
+
+// Locator returns a locator for the given resource
+func (r *ResourceGroup) Locator(api *API) *ResourceGroupLocator {
+	for _, l := range r.Links {
+		if l["rel"] == "self" {
+			return api.ResourceGroupLocator(l["href"])
+		}
+	}
+	return nil
+}
+
+//===== Locator
+
+// ResourceGroupLocator exposes the ResourceGroup resource actions.
+type ResourceGroupLocator struct {
+	Href
+	api *API
+}
+
+// ResourceGroupLocator builds a locator from the given href.
+func (api *API) ResourceGroupLocator(href string) *ResourceGroupLocator {
+	return &ResourceGroupLocator{Href(href), api}
+}
+
+//===== Actions
+
+// POST /api/resource_groups
+//
+// Creates a ResourceGroup.
+// Required parameters:
+// resource_group
+func (loc *ResourceGroupLocator) Create(resourceGroup *ResourceGroupParam) (*ResourceGroupLocator, error) {
+	var res *ResourceGroupLocator
+	if resourceGroup == nil {
+		return res, fmt.Errorf("resourceGroup is required")
+	}
+	var params rsapi.APIParams
+	var p rsapi.APIParams
+	p = rsapi.APIParams{
+		"resource_group": resourceGroup,
+	}
+	uri, err := loc.ActionPath("ResourceGroup", "create")
+	if err != nil {
+		return res, err
+	}
+	req, err := loc.api.BuildHTTPRequest(uri.HTTPMethod, uri.Path, APIVersion, params, p)
+	if err != nil {
+		return res, err
+	}
+	resp, err := loc.api.PerformRequest(req)
+	if err != nil {
+		return res, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode < 200 || resp.StatusCode > 299 {
+		respBody, _ := ioutil.ReadAll(resp.Body)
+		sr := string(respBody)
+		if sr != "" {
+			sr = ": " + sr
```

Comment:
why do this instead of just doing something like "invalid response status:%s body:%s" ?